### DISCUSSION TITLE
Welcome to django-admin-autocomplete-filter Discussions!

### DISCUSSION THREAD

**Discussion by farhan0581** [farhan0581/django-admin-autocomplete-filter#87] [Announcements]
## 👋 Urgent Improvement Discussion !
  Lets use this space to discuss about the urgent priority features and updates required for this project:
  * Context : The versions of Django and Python for which I created this project are dead now, so need to update this.
  * Should we update this project to support Django 3 or 4 ?
  

  Please help me to share or reshare the existing prs for this project in order to update this project. Really sorry for the delay, lets help each other to improve this 💪.

**Comment by andyzickler** [comment]
Here's what i would recommend for a strategy:
* Only support the currently supported versions of Django. 
Right now thats 3.2, 4.0, and 4.1. In April 4.2 will be released and 4.0 will be dropped
* Support the last 2ish versions of Python that Django supports. So Python 3.10 and 3.11
* Use Tox to test multiple all combinations of django and python versions
* Treat any deprecation warnings as errors in the tests and prioritize fixing them.
This way we can be assured that the next version of Django won't cause any issues and people who use this module can update django without fear
* Setup GitHub actions to run Tox on pull requests. 

Not everything here needs to be done in one step. It might be good to start with setting up Tox and GitHub actions first with the currently supported old versions of Django, and then later updating to Django 3.2 and 4.

I helped setup tox and GitHub actions on another django module that you can use for reference. Here is [the tox.ini file](https://github.com/crucialfelix/django-ajax-selects/blob/develop/tox.ini) and [the github workflow](https://github.com/crucialfelix/django-ajax-selects/blob/develop/.github/workflows/tests.yml)

**Reply by farhan0581** [reply]
This seems to be a good action plan, will look into this.

**Comment by farhan0581** [comment]
I think next we can plan to support Django version 3.2 and post that we can target Django 4 support

**Reply by SiberMix** [reply]
When for django 4?)

**Comment by jsutaria** [comment]
Any updates on allowing for autocomplete multiselect in listfilters?

**Reply by farhan0581** [reply]
I am currently planning to fix this for django version 3.2
Will share the updates here

**Comment by nemesifier** [comment]
I see the project has not been maintained in a while. Are you not using it anymore in your projects @farhan0581?
